### PR TITLE
tests(remix-react): add failing test for default `Form` action

### PR DESCRIPTION
**Pull request by ionut-botizan**
Issue: #3133

The default `<Form />` action should match the current location. Currently, only the `path` is preserved, but the `search` parameters are lost.

E.g. For the `/login?redirectTo=/profile` location and a `<Form method="post" />` element, the generated form node looks like
`<form method="post" action="/login">`.

### CODE REVIEW

File: integration/bug-report-test.ts
Comment (chaance) [inline integration/bug-report-test.ts:50]: This test file is used as a template. Please move this test to `integration/form-test.ts`

File: packages/remix-react/components.tsx
Comment (chaance) [inline packages/remix-react/components.tsx:968]: This should check if `action === undefined` instead. If the action is explicitly set to `.` it will be resolved to strip the search, which is the expected behavior.

We only want to use the location's search if the action is left undefined in the Form. This means we need to remove the default value of the action in `FormImpl` as well.

See https://github.com/remix-run/remix/issues/3133#issuecomment-1128776268

**Review by chaance — CHANGES_REQUESTED** [review]
A few changes needed before we can merge. If an action value of `.` is provided, it is correct to strip the search params when resolving the route path. We only need to preserve the location's search if no action is provided.

File: integration/bug-report-test.ts
Comment (ionut-botizan) [reply integration/bug-report-test.ts:50]: Oopsie! 😊

File: packages/remix-react/components.tsx
Comment (ionut-botizan) [reply packages/remix-react/components.tsx:968]: Got it!

**Comment by maranomynet** [comment]
Please someone resolve and merge this PR.

Always having to do this feels so daft.

```ts
const { pathname, search } = useLocation();

return <Form action={pathname+search}
```

**Comment by ionut-botizan** [comment]
Hey guys! Are there any other changes or additions that you would like me to provide? I think I did everything that was asked in the initial review...
I don't mean to bother you. I realize there are more important issues out there I was just thinking that maybe you missed the fact that I responded to the review. 🙂

**Comment by chaance** [comment]
Hi @ionut-botizan, sorry for the radio silence on this one. We actually have duplicated efforts a bit on this bug. We are building Remix's form components and hooks into React Router, so @brophdawg11 and I have been working in tandem on that separately. I think I must have forgotten about your work and started my own over in https://github.com/remix-run/remix/pull/3697. Very sorry about that. 

To clean things up, I'm going to close this one. But I added you as a co-committer for [`783c435` (#3697)](https://github.com/remix-run/remix/pull/3697/commits/783c43573f37bb89e1cf25be5ec87b72aee5186f) and made sure your handle is included in the contributor list. I definitely want to make sure you're credited with identifying the problem and proposing the initial fix!

**Pull request closed without merging**
